### PR TITLE
fix: console.log formatting

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

Closes https://github.com/foundry-rs/foundry/issues/8700

## Solution

There are several issues:
1. We attempt parsing `FormatSpec` and thus consume some characters even if there are no values to format. Those characters are not pushed to resulted string. Solution is to skip parsing `FormatSpec` if there are no characters left.
2. On parser failure, we only push a single character instead of all consumed. Solution is to return `String` in `Err` variant with all consumed chars.
3. There are probably some more issues because of `&mut chars` being wrapped into `chain` and `Peekable` thus making it a bit hard to follow on how many times it was actually advanced. Solution is to use a single iterator everywhere

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
nice!!

**Comment by mds1** [comment]
That was fast, thank you!
